gallery/gallery/lib/demos/material/dialog_demo.dart

```diff
@@ -124,42 +124,72 @@ class DialogDemo extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      key: _scaffoldKey,
-      appBar: AppBar(
-        title: Text(_title(context)),
-      ),
-      body: Center(
-        child: RaisedButton(
-          child: Text(GalleryLocalizations.of(context).dialogShow),
-          onPressed: () {
-            switch (type) {
-              case DialogDemoType.alert:
-                _showAlertDialog(context);
-                break;
-              case DialogDemoType.alertTitle:
-                _showAlertDialogWithTitle(context);
-                break;
-              case DialogDemoType.simple:
-                _showSimpleDialog(context);
-                break;
-              case DialogDemoType.fullscreen:
-                Navigator.push<void>(
-                  context,
-                  MaterialPageRoute(
-                    builder: (context) => _FullScreenDialogDemo(),
-                    fullscreenDialog: true,
-                  ),
-                );
-                break;
-            }
-          },
-        ),
-      ),
+    return Navigator(
+      // Adding [ValueKey] to make sure that the widget gets rebuilt when
+      // changing type.
+      key: ValueKey(type),
+      onGenerateRoute: (settings) {
+        return _NoAnimationMaterialPageRoute<void>(
+          builder: (context) => Scaffold(
+            key: _scaffoldKey,
+            appBar: AppBar(
+              automaticallyImplyLeading: false,
+              title: Text(_title(context)),
+            ),
+            body: Center(
+              child: RaisedButton(
+                child: Text(GalleryLocalizations.of(context).dialogShow),
+                onPressed: () {
+                  switch (type) {
+                    case DialogDemoType.alert:
+                      _showAlertDialog(context);
+                      break;
+                    case DialogDemoType.alertTitle:
+                      _showAlertDialogWithTitle(context);
+                      break;
+                    case DialogDemoType.simple:
+                      _showSimpleDialog(context);
+                      break;
+                    case DialogDemoType.fullscreen:
+                      Navigator.push<void>(
+                        context,
+                        MaterialPageRoute(
+                          builder: (context) => _FullScreenDialogDemo(),
+                          fullscreenDialog: true,
+                        ),
+                      );
+                      break;
+                  }
+                },
+              ),
+            ),
+          ),
+        );
+      },
     );
   }
 }
 
+/// A MaterialPageRoute without any transition animations.
+class _NoAnimationMaterialPageRoute<T> extends MaterialPageRoute<T> {
```

Comment:
What was the need to do this @perclasson ?

Reply:
Otherwise on the macOS at least, when I change the demo type it animates in from the left. Looks a bit different compared to the other demos.